### PR TITLE
Issue 256 bitwise operators

### PR DESCRIPTION
**Pull request by w84thesun**
Closes https://github.com/FerretDB/FerretDB/issues/256

### CODE REVIEW

File: cmd/ferretdb/main.go [hunk 1]
```diff
@@ -20,17 +20,16 @@ import (
...
+	"syscall"
 
 	"github.com/FerretDB/FerretDB/internal/clientconn"
 	"github.com/FerretDB/FerretDB/internal/pg"
 	"github.com/FerretDB/FerretDB/internal/util/debug"
 	"github.com/FerretDB/FerretDB/internal/util/logging"
 	"github.com/FerretDB/FerretDB/internal/util/version"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/expfmt"
+	"go.uber.org/zap"
```
Comment (seeforschauer) [inline cmd/ferretdb/main.go:32]: import sorting?

File: cmd/ferretdb/main.go [hunk 2]
```diff
@@ -83,7 +82,7 @@ func main() {
 		logger.Sugar().Warn("The current TLS implementation is not secure.")
 	}
 
-	ctx, stop := signal.NotifyContext(context.Background(), unix.SIGTERM, unix.SIGINT)
```
Comment (seeforschauer) [inline cmd/ferretdb/main.go:86]: why have you changed from unix to syscall?

File: cmd/ferretdb/main.go (hunk 1 above)
Comment (w84thesun) [reply cmd/ferretdb/main.go]: Forgot to setup gci. Fixed

**Comment by AlekSi** [comment]
Let's use draft status for draft PRs

File: cmd/ferretdb/main.go (hunk 2 above)
Comment (w84thesun) [reply cmd/ferretdb/main.go]: Didn't know that syscall is deprecated.

File: .golangci.yml [hunk 3]
```diff
@@ -51,6 +51,11 @@ linters-settings:
   gosimple:
     go: "1.18"
     checks: ["all"]
+  gci:
+    sections:
+      - standard
+      - default
+      - prefix(github.com/FerretDB/FerretDB)
```
Comment (AlekSi) [inline .golangci.yml:58]: We have that section above already (merged in another PR)

File: internal/types/binary.go [hunk 4]
```diff
@@ -35,3 +41,42 @@ type Binary struct {
 	Subtype BinarySubtype
 	B       []byte
 }
+
+func BinaryFromArray(values *Array) (*Binary, error) {
```
Comment (AlekSi) [inline internal/types/binary.go:45]: Our linter for that is broken right now (should be fixed by 1.18 release), but: every exported function should have a documentation comment.

It is somewhat hard to understand what conversion from array to binary even mean

**Review by AlekSi — COMMENTED** [review]
We also need dance tests

File: internal/handlers/jsonb1/where.go [hunk 5]
```diff
@@ -60,6 +62,15 @@ func scalar(v any, p *pg.Placeholder) (sql string, args []any, err error) {
...
 		}
+	case types.Binary:
+		sql = fmt.Sprintf(`)::integer::bit(%d) # `+p.Next()+` )::integer = 0`, 8)
+		bit := pgtype.Bit{
+			Bytes:  v.B,
+			Len:    8,
+			Status: pgtype.Present,
+		}
+
+		arg, _ = bit.Value()
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/where.go:73]: always check errors

File: internal/handlers/jsonb1/where.go [hunk 6]
```diff
@@ -60,6 +62,15 @@ func scalar(v any, p *pg.Placeholder) (sql string, args []any, err error) {
 		if options != "" {
 			arg = "(?" + options + ")" + v.Pattern
 		}
+	case types.Binary:
+		sql = fmt.Sprintf(`)::integer::bit(%d) # `+p.Next()+` )::integer = 0`, 8)
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/where.go:66]: interesting escaping, will use in my PRs :)

File: internal/handlers/jsonb1/where.go (hunk 5 above)
Comment (w84thesun) [reply internal/handlers/jsonb1/where.go]: fixed

File: .golangci.yml (hunk 3 above)
Comment (w84thesun) [reply .golangci.yml:58]: Going to drop this commit.

**Review by seeforschauer — CHANGES_REQUESTED** [review]
dance tests failing

**Pull request closed without merging**